go/vitess/api_client.go

```diff
@@ -0,0 +1,49 @@
+package vitess
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Tablet represents information about a running instance of vttablet.
+type Tablet struct {
+	MysqlHostname string `json:"mysql_hostname,omitempty"`
+	MysqlPort     int32  `json:"mysql_port,omitempty"`
+}
+
+var httpClient = http.Client{
+	Timeout: 1 * time.Second,
+}
+
+func constructAPIURL(api string, keyspace string, shard string) (url string) {
```

Comment:
Are you connecting to consul and directly reading the data from there?  That seems like it will miss tablets that are in different cells than the consul service you're attached to.  It seems like this could break if vitess decides to change their data structure, and it won't work if someone decides to use another key storage system like `etcd` or `zookeeper` which are also supported by vitess.

I have an example of how to call the vitess client through gRPC here that may be helpful as an alternative: https://github.com/github/db-scripts/blob/master/vitess/client.go

Reply:
@gtowey `freno` does not read data from Consul here. It accesses a new `vtctld` Web API endpoint. That endpoint only exists in our internal Vitess code right now, but once tested for a few days it will be shared upstream, at which point I will drop a link in this public repo.

Reply:
Ah, I wasn't aware `vtctld` had a web api -- I thought it was all through grpc.  Well then, carry on =)